src/config.js

```diff
@@ -1,20 +1,120 @@
+ * @param  {Number} [userConfig.port] - Port for your custom service
+ * @param  {Boolean} [userConfig.secure ]- Set to true for your custom service if you want a (TLS) secured connection
+ * @throws Invalid Configurations provided for custom service
+ * @throws Invalid settings provided for senderEmail, senderPasword, or mailService
+ * @returns {config} configuration that has been set
+ */
+function setConfig(
+  userConfig = {
+    hostingURL: '',
+    mailService: 'Gmail',
+    senderEmail: '',
+    senderPassword: '',
+    host: '',
+    port: 465,
+    secure: false,
+  }
+) {
+  config = {}
+  // config.hostingURL = (validator.isURL(userConfig.hostingURL || process.env.HOSTING_URL))?userConfig.hostingURL || process.env.HOSTING_URL: '';
+  config.hostingURL = userConfig.hostingURL || process.env.HOSTING_URL;
+  // DONE: Better  validation
+  
+  if ((userConfig.mailService || process.env.MAIL_SERVICE) === 'Custom') {
+    if ( validator.isFQDN(userConfig.host || process.env.MAIL_HOST) && validator.isPort(`${userConfig.port || process.env.MAIL_PORT}`) && !validator.isEmpty(userConfig.senderEmail || process.env.MAIL_USER) && !validator.isEmpty(userConfig.senderPassword || process.env.MAIL_PASSWORD) && validator.isBoolean(`${userConfig.secure || process.env.MAIL_SECURE}`) ) {
+      config.mailService = 'Custom'
+      config.host = userConfig.host || process.env.MAIL_HOST;
+      config.port = userConfig.port || process.env.MAIL_PORT;
+      config.secure = userConfig.secure || process.env.MAIL_SECURE;
+      config.auth = {
+        user: userConfig.senderEmail || process.env.MAIL_USER,
+        pass: userConfig.senderPassword || process.env.MAIL_PASSWORD,
+      };
+      return config
+    } else {
+      throw new Error('Invalid Configurations provided for custom service');
+    }
+  }
+  config.mailService = (userConfig.mailService || process.env.MAIL_SERVICE)? userConfig.mailService || process.env.MAIL_SERVICE : 'Gmail';
+  if (validator.isEmail(userConfig.senderEmail || process.env.MAIL_USER) && !validator.isEmpty(userConfig.senderPassword || process.env.MAIL_PASSWORD)) {
+    config.senderEmail = userConfig.senderEmail || process.env.MAIL_USER
+    // deepcode ignore WrongNumberOfArgs: Because it is wrong
+    config.senderPassword = userConfig.senderPassword || process.env.MAIL_PASSWORD
   } else {
-    throw new Error('senderEmail, senderPasword, not provided')
+    throw new Error('Invalid settings provided for senderEmail, senderPasword, or mailService')
   }
+  return config
 }
 
+/**
+ * @returns {config} Config
+ */
 const getConfig = () => {
   return config
 }
 
-module.exports = { getConfig, setConfig }
+/**
+ * @param {config} [config]
+ * @returns {mailer.createTransport}
+ */
+const getTransporter = (config = {}) => {
+  if (Object.keys(config).length !== 0 && config.constructor === Object) {
+    setConfig(config)
+  }
+  if (config.mailService !== 'Custom') {
```

Comment:
What provider might fall under `Custom`?

Reply:
In custom, any and all providers can be configured, there will be no restriction.